src/server/frontend_xwayland/xwayland_clipboard_provider.cpp

```diff
@@ -52,38 +54,139 @@ auto create_selection_window(mf::XCBConnection const& connection) -> xcb_window_
     return selection_window;
 }
 
-class ReadError : public std::runtime_error
+void send_selection_notify(
+    mf::XCBConnection& connection,
+    xcb_timestamp_t time,
+    xcb_window_t requestor,
+    xcb_atom_t property,
+    xcb_atom_t selection,
+    xcb_atom_t target)
```

Comment:
Needed to be broken out of `XWaylandClipboardProvider` because both that and `SelectionSender` need to call it.